tiatoolbox/utils/transforms.py

```diff
@@ -83,7 +83,7 @@ def imresize(img, scale_factor=None, output_size=None, interpolation="optimise")
     """
     if scale_factor is None and output_size is None:
         raise TypeError("One of scale_factor and output_size must be not None.")
-    if scale_factor is not None:
+    elif scale_factor:
```

Comment:
Why elif? this would never be reached if raise is hit. I think this should be if not elif.